os/fs/vfs/fs_open.c

```diff
@@ -164,11 +168,10 @@ int open(const char *path, int oflags, ...)
 		/* Get the file descriptor of the opened character driver proxy */
 		fd = block_proxy(path, oflags);
 		if (fd < 0) {
-			ret = -fd;
+			ret = fd;
```

Comment:
To return a negative fd is meaningful for debugging..? It seems that to return a specific errno value (negated) is better.

Reply:
If `block_proxy` returns a negative value, it may be ENOMEM.